Consolidate release notes for 3.6 (#23814)
matplotlib/matplotlib: doc/api/prev_api_changes/api_changes_3.6.0/behaviour.rst

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When showing an image using `~matplotlib.axes.Axes.imshow` that has more than
:math:`2^{24}` columns or :math:`2^{23}` rows, the image will now be
downsampled to below this resolution before being resampled for display by the
AGG renderer. Previously such a large image would be shown incorrectly. To
prevent this downsampling and the warning it raises, manually downsample your
data before handing it to `~matplotlib.axes.Axes.imshow`.

New *algorithm* keyword argument to ``contour`` and ``contourf``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The contouring functions `~matplotlib.axes.Axes.contour` and
`~matplotlib.axes.Axes.contourf` have a new keyword argument *algorithm* to
control which algorithm is used to calculate the contours. There is a choice of
four algorithms to use, and the default is to use ``algorithm='mpl2014'`` which
is the same algorithm that Matplotlib has been using since 2014.

Other possible values of the *algorithm* keyword argument are ``'mpl2005'``,
``'serial'`` and ``'threaded'``; see the `ContourPy documentation
<https://contourpy.readthedocs.io>`_ for further details.

> Default date limits changed to 1970-01-01 to 1970-01-02

Reviewer: The double "to" is slightly confusing. Maybe

```suggestion
Default date limits changed to 1970-01-01 - 1970-01-02
```

`.Axes.get_yticklabels`) would only return empty strings unless a draw had
already been performed. Now the ticks and their labels are updated when the
labels are requested.

Added *markerfacecoloralt* parameter to ``Axes.errorbar``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The *markerfacecoloralt* parameter is now passed to the line plotter from
`.Axes.errorbar`. The documentation now accurately lists which properties are
passed to `.Line2D`, rather than claiming that all keyword arguments are passed
on.

Warning when scatter plot color settings discarded
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When making an animation of a scatter plot, if you don't set *c* (the color
value parameter) when initializing the artist, the color settings are ignored.
`.Axes.scatter` now raises a warning if color-related settings are changed
without setting *c*.

3D ``contourf`` polygons placed between levels
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The polygons used in a 3D `~.Axes3D.contourf` plot are now placed halfway
between the contour levels, as each polygon represents the location of values
that lie between two levels.

> Axes title now avoids y-axis offset
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> Previously, Axes titles could overlap the y-axis offset text, which is often in
> the upper left corner of the axes. Now titles are moved above the offset text
> if overlapping, and automatic positioning is in effect (i.e. if *y* in
> `.Axes.set_title` is *None* and :rc:`axes.titley` is also *None*).

Reviewer 2: ```suggestion
Axes title now avoids y-axis offset
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Previously, Axes titles could overlap the y-axis offset text, which is often in
the upper left corner of the axes. Now titles are moved above the offset text
if overlapping when automatic title positioning is in effect (i.e. if *y* in
`.Axes.set_title` is *None* and :rc:`axes.titley` is also *None*).
```

require the *renderer* keyword argument, saving users from having to query it
from ``fig.canvas.get_renderer``. If the *renderer* keyword argument is not
supplied, these methods first check if there is a cached renderer from a
previous draw and use that. If there is no cached renderer, then the methods
will use ``fig.canvas.get_renderer()`` as a fallback.

``FigureFrameWx`` constructor, subclasses, and ``get_canvas``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``FigureCanvasWx`` constructor gained a *canvas_class* keyword-only
parameter which specifies the canvas class that should be used. This parameter
will become required in the future. The ``get_canvas`` method, which was
previously used to customize canvas creation, is deprecated. The
``FigureFrameWxAgg`` and ``FigureFrameWxCairo`` subclasses, which overrode
``get_canvas``, are deprecated.

``FigureFrameWx.sizer``
~~~~~~~~~~~~~~~~~~~~~~~

... has been removed. The frame layout is no longer based on a sizer, as the
canvas is now the sole child widget; the toolbar is now a regular toolbar added
using ``SetToolBar``.

Incompatible layout engines raise
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> ``tight_layout`` and ``constrained_layout`` are incompatible if a colorbar has
> been added to the figure. Invoking the incompatible layout engine used to warn,
> but now raises with a `RuntimeError`.
> 

Reviewer 2: Sorry, this was my wording, but if we can clarify:

```suggestion
You cannot switch between ``tight_layout`` and ``constrained_layout`` if a colorbar has
already been added to a figure. Invoking the incompatible layout engine used to warn,
but now raises with a `RuntimeError`.
```

``CallbackRegistry`` raises on unknown signals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When Matplotlib instantiates a `.CallbackRegistry`, it now limits callbacks to
the signals that the registry knows about. In practice, this means that calling
`~.FigureCanvasBase.mpl_connect` with an invalid signal name now raises a

> `ValueError`.

Reviewer 3: Markup for builtin exception classes is inconsistent between e.g. here and the next entry.

Author: Fixed.
